### DISCUSSION TITLE
How to update connection string for existing connection pool setups

### DISCUSSION THREAD

**Discussion by mtuncer** [jeremyevans/sequel#2164] [General]
I am connecting to a postgres database. user password may be changed/rotated at arbitrary times by an external entity.  

all my connections are setup during `Sequal.connect` during client service startup. If the connection is dropped/or session timed out for some reason, reconnection is not possible because connection string is no longer valid.

my ask is
- could there be a call back to notify app that re-connection fails, and
- a way to update the connection string when a password is changed

just the second one could be enough to move forward, I can put up a periodic tasks to update the connection string on the client service side if needed.

**Comment by jeremyevans** [comment]
You could pass a mutable string, and use String#replace to update it if the information needs to be changed (a mutable hash instead of a mutable string is also possible). Not exactly good design, but having authentication information change without your control is not good design either. :) 

Note that if a connection to the database fails, Sequel raises a Sequel::DatabaseConnectionError exception, so you could rescue that.

**Reply by mtuncer** [reply]
Thanks, I should have been more clear on the request.

I want to use azure entraid authentication which uses `tokens` instead of passwords. each token has a validity period, so I need to fetch a valid token before reconnecting.

**Comment by jeremyevans** [comment]
I added 8a189713dc562a75521fd2bfbfc72c64b68a04e3 which I think should be able to handle your use case.  Could you try it out and provide feedback?
